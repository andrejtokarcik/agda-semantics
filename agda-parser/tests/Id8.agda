
-- This module introduces implicit arguments.

module Id8 where

id8 : {A : Set} -> A -> A
id8 = \{A} x -> x        -- this doesn't work since the type checker assumes
                            -- that the implicit A has been has been omitted in
                            -- the left-hand side (as in id6).
