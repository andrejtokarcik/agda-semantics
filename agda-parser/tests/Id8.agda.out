<k>
    whnf ( Fun ( "t1" ) ) ~> nf ( HOLE )
</k>
<dbg>
    0
</dbg>
<mgu>
    subst(.KList)
</mgu>
<ctxstack>
    .List
</ctxstack>
<fenv-stack>
    .List
</fenv-stack>
<sig>
    Fun ( "id8" ) |-> { Var ( "A" ) : Set ( 0 ) } -> ( Var ( "x" ) : Var ( "A"
       ) ) -> Var ( "A" )
</sig>
<ctx>
    .Map
</ctx>
<meta>
    .Map
</meta>
<fenv>
    .Map
</fenv>
<ncat>
    Name ( "id8" ) |-> Fun ( "id8" )
</ncat>
<fun>
    Fun ( "id8" ) |-> 'FunClauses((#symFunClauses(20) { #symExpr(2) }) = (\ { 
      Var ( "A" ) : #symExpr(16) } -> \ ( Var ( "x" ) : #symExpr(10) ) -> Var 
      ( "x" ) { #symExpr(13) }))
</fun>
